src/repr/src/row/encoding.rs

```diff
@@ -242,20 +245,23 @@ pub trait DatumToPersist {
 /// `FnOnce<T: DatumToPersist>() -> R`
 ///
 /// This saves us from needing another `enum_dispatch` for [DatumToPersist].
-pub trait DatumToPersistFn<R> {
+pub trait DatumToPersistFn<'a, R: 'a> {
     fn call<T: DatumToPersist>(self) -> R
     where
-        for<'a> DatumDecoder<'a>: From<DataRef<'a, T>>,
-        for<'a> DatumEncoder<'a>: From<DataMut<'a, T>>;
+        DatumDecoder<'a>: From<DataRef<'a, T>>,
+        DatumEncoder<'a>: From<DataMut<'a, T>>;
 }
```

Comment:
I needed to make this change because otherwise once I added `Array` to datum encoder Rust complained about something like:
```
DatumEncoder<'0> needs to implement From<DataRef<'0, Array<'_>> for all lifetimes '0, but it only implements From<DataRef<'1, Array<'1>> for a specific lifetime '1
```

I tried a few different ways to fix this, adding a second lifetime to `DatumEncoder` and using generic associated types, but this was the only way to do it